### ISSUE TITLE
Retries not working as expected

### ISSUE THREAD

**Issue by BenGable** [guilatrova/gracy#30]
Hi,

I've been using Gracy in production for a little while now and it's been great, so thank you!

The API I'm using likes to take far too long to respond, so timeouts are a pretty regular occurence and I've noticed two related issues.

The first is that retrying specific exceptions does not seem to work at all (retry_on=None works fine).

```python
import asyncio
from gracy import BaseEndpoint, Gracy, GracyConfig, LogEvent, LogLevel, GracefulRetry
from httpx import ReadTimeout

class TestEndpoint(BaseEndpoint):
    OK = "/200/"

class GracefulTestAPI(Gracy[str]):
    class Config:
        BASE_URL = "https://httpstat.us/"
        REQUEST_TIMEOUT = 5.0
        SETTINGS = GracyConfig(
            log_request=LogEvent(LogLevel.DEBUG),
            log_response=LogEvent(LogLevel.INFO, "{URL} took {ELAPSED}"),
            parser={
                "default": lambda r: r.json()
            },
            retry=GracefulRetry(
                delay=1,
                max_attempts=5,
                delay_modifier=3,
                retry_on={
                    ReadTimeout,
                },
                log_before=LogEvent(LogLevel.WARNING),
                log_after=LogEvent(LogLevel.WARNING),
                log_exhausted=LogEvent(LogLevel.ERROR),
                behavior="break",
            ),
        )

    async def test(self):
        return await self.get(TestEndpoint.OK, params={"sleep": "6000"})  # <- force a timeout

testapi = GracefulTestAPI()

async def main():
    test_resp = await testapi.test()
    print(test_resp)

asyncio.run(main())
```

In the above example the ReadTimeout is not retried at all. I think I've traced it to this bit of code in GracyConfig.should_retry (line 562):

```python
for maybe_exc in retry_on:
    if inspect.isclass(maybe_exc) and isinstance(
        req_or_validation_exc, maybe_exc
    ):
        return True
```
As far as I can tell the exception will always be wrapped in a `GracyRequestFailed` at this point, so `req_or_validation_exc` will never match any of the listed exceptions to retry. So I think it should instead be checking the `original_exc` attribute against `maybe_exc`.

The second - much more minor - issue I noticed is that the exception message reads "The request for [GET] https://httpstat.us/200/ never got a response due to  " (i.e. no exception info). This is because httpx.ReadTimeout (and possibly all httpx exceptions?) has no message (related issue [here](https://github.com/encode/httpx/issues/683)). Maybe instead of `str(original_exc)` GracyRequestFailed could use the repr (possibly just for httpx exceptions?) instead so there's a useful output when logging?

**Comment by guilatrova** [comment]
Thanks for notifying. I'll check this and push a fix.

**Comment by guilatrova** [comment]
Hi Ben.

The fix has been released and it's available on [version 1.32.1](https://pypi.org/project/gracy/1.32.1/).

Thank you very much for the detailed issue description, it helped a lot to reproduce, debug, and fix it.

Also, I'm very glad to hear you've been using it in production for a while! :smile: I'll reach out through email (I hope you don't mind) to ask about how has been your usage/experience.

Thank you very much for using and supporting Gracy!
